sklearn/utils/tests/test_utils.py

```diff
@@ -472,8 +472,12 @@ def test_safe_indexing_pandas_no_settingwithcopy_warning():
 
     X = pd.DataFrame({"a": [1, 2, 3], "b": [3, 4, 5]})
     subset = _safe_indexing(X, [0, 1], axis=0)
+    if hasattr(pd.errors, "SettingWithCopyWarning"):
+        SettingWithCopyWarning = pd.errors.SettingWithCopyWarning
+    else:
+        SettingWithCopyWarning = pd.core.common.SettingWithCopyWarning
```

Comment:
A comment with the pandas version would be nice here so that we know when we can remove it . This line is for pandas < 1.5 I think.